core/corerepo/pinning.go

```diff
@@ -61,10 +63,12 @@ func Pin(n *core.IpfsNode, paths []string, recursive bool) ([]u.Key, error) {
 }
 
 func Unpin(n *core.IpfsNode, paths []string, recursive bool) ([]u.Key, error) {
+	// TODO(cryptix): do we want a ctx as first param for (Un)Pin() as well, just like core.Resolve?
```

Comment:
most likely, yeah. I'd also be inclined to move Unpin towards taking only a single path and moving the iteration logic upwards.